src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -306,11 +308,33 @@ def _set_rsa_parameters(dest, src):
     return client.import_key(
         vault_base_url, key_name, key_obj, destination == 'hsm', key_attrs, tags)
 
-# pylint: disable=unused-argument
-def download_secret(client, vault_base_url, secret_name, file_path, file_encoding='utf8',
-                    secret_version='', decode_binary=None):
+def download_secret(client, vault_base_url, secret_name, file_path, encoding=None,
+                    secret_version=''):
+    """ Download a secret from a KeyVault. """
+    if os.path.isfile(file_path):
+        raise CLIError("File '{}' already exists.".format(file_path))
+
     secret = client.keyvault.get_secret(vault_base_url, secret_name, secret_version)
-    raise CLIError('TODO: implement')
+    encoding = encoding or secret.tags.get('file-encoding', 'utf-8')
+    secret_value = secret.value
+
+    try:
+        if encoding in secret_text_encoding_values:
+            with open(file_path, 'w') as f:
+                f.write(secret_value)
+        else:
+            if encoding == 'base64':
+                import base64
+                decoded = base64.b64decode(secret_value)
+            elif encoding == 'hex':
+                import binascii
+                decoded = binascii.unhexlify(secret_value)
+
+            with open(file_path, 'wb') as f:
+                f.write(decoded)
+    except Exception: # pylint: disable=broad-except
+        if os.path.isfile(file_path):
+            os.remove(file_path)
```

Comment:
You catch the exception which is fine but should you then raise CLIError after you remove the file?
Otherwise I think the command would return an exit code of 0 and it'd look like it downloaded fine.